meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,723 +11,903 @@ import {
+				const deviceTypeFromMappedLayer: TSR.DeviceType | undefined = mappedLayer?.device
+
+				const routeDeviceType: TSR.DeviceType | undefined =
+					route.routeType === StudioRouteType.REMAP
+						? route.deviceType
+						: route.mappedLayer
+						? deviceTypeFromMappedLayer
+						: route.deviceType
+
+				const routeMappingSchema = manifest[(routeDeviceType ?? route.remapping?.device) as TSR.DeviceType]
+
+				const rawMappingTypeOptions = Object.entries<JSONSchema>(routeMappingSchema?.mappingsSchema || {})
+				const mappingTypeOptions = rawMappingTypeOptions.map(([id, entry], i) =>
+					literal<DropdownInputOption<string | number>>({
+						value: id + '',
+						name: entry?.title ?? id + '',
+						i,
+					})
 				)
-			}
 
-			return _.map(this.props.studio.routeSets, (routeSet: StudioRouteSet, routeId: string) => {
 				return (
-					<React.Fragment key={routeId}>
-						<tr
-							className={ClassNames({
-								hl: this.isItemEdited(routeId),
-							})}
-						>
-							<th className="settings-studio-device__name c2">{routeId}</th>
-							<td className="settings-studio-device__id c3">{routeSet.name}</td>
-							<td className="settings-studio-device__id c4">{routeSet.exclusivityGroup}</td>
-							<td className="settings-studio-device__id c2">{routeSet.routes.length}</td>
-							<td className="settings-studio-device__id c2">
-								{routeSet.active ? <span className="pill">{t('Active')}</span> : null}
-							</td>
-
-							<td className="settings-studio-device__actions table-item-actions c3">
-								<button className="action-btn" onClick={() => this.editItem(routeId)}>
-									<FontAwesomeIcon icon={faPencilAlt} />
-								</button>
-								<button className="action-btn" onClick={() => this.confirmRemove(routeId)}>
-									<FontAwesomeIcon icon={faTrash} />
-								</button>
-							</td>
-						</tr>
-						{this.isItemEdited(routeId) && (
-							<tr className="expando-details hl">
-								<td colSpan={6}>
-									<div className="properties-grid">
-										<label className="field">
-											<LabelActual label={t('Route Set ID')} />
-											<EditAttribute
-												modifiedClassName="bghl"
-												attribute={'routeSets'}
-												overrideDisplayValue={routeId}
-												obj={this.props.studio}
-												type="text"
-												collection={Studios}
-												updateFunction={this.updateRouteSetId}
-												className="input text-input input-l"
-											></EditAttribute>
-										</label>
-										<label className="field">
-											<LabelActual label={t('Active')} />
+					<div className="route-sets-editor mod pan mas" key={index}>
+						<button className="action-btn right mod man pas" onClick={() => confirmRemoveRoute(routeSetId, route)}>
+							<FontAwesomeIcon icon={faTrash} />
+						</button>
+						<div className="properties-grid">
+							<label className="field">
+								<LabelActual label={t('Original Layer')} />
+								<EditAttribute
```

Comment:
These `EditAttribute` components won't work. They don't handle the `ObjectWithOverrides`.
This usage is still modifying the property names prior to the change. They will need to be changed out to be components like being used elsewhere in this file, using `overrideHelper` to help with writing changes to the correct place

Reply:
Will do.
This was taken from Mapping.tsx so maybe we should change it there too?
https://github.com/bbc/sofie-core/blob/7a4e1c7f54c75a37ea6a03683820dd8b25b440e6/meteor/client/ui/Settings/Studio/Mappings.tsx#L416C1-L426C2

Reply:
I don't see an `EditAttribute` in Mapping.tsx

Reply:
Sorry I looked at the` <TextInputControl`
The one here is part of the `RenderRoutes` mentioned in the next comment.
(The part I've overlooked)